### DISCUSSION TITLE
Feedback on adding nullability annotations

### DISCUSSION THREAD

**Discussion by paulbakker** [graphql-java/graphql-java#3630] [General]
Hi folks, just some learnings from upgrading to GraphQL Java 22.

Nullability annotations were added to several methods. While this looks like a small change (and a good one), we did see issues caused by this. I'm just bringing this up to keep in mind in the future, and it doesn't mean such changes shouldn't be made.

Builds started failing because of the new nullability in two scenarios:
1. Kotlin apps - Since kotlin is strict about null checks, compilation actually fails after adding these annotations.
2. SpotBugs - For Java apps that use SpotBugs, there might be more enforcement on nullability checks.

Because of the failing builds, it's sort of a breaking change, but probably not considered that way.

Again, not a big deal, just something to be aware of.

**Comment by bbakerman** [comment]
yeah we under estimated the "breaking change ness" of this PR.  An external contributor wanted to add an extra "improvement" on the non nullability of key methods in DataFetcherEnvironment

https://github.com/graphql-java/graphql-java/pull/3582

They were always non null in effect but not indicated as such.  We don't have extensive coverage on annotations (although we have increased the use of them over time)

So this betterness change ends up breaking Kotlin compilation.

We will be more deligent in the future on this kinda of change.

**Reply by dondonz** [reply]
I'll edit the release notes to highlight this, thanks @paulbakker

**Reply by paulbakker** [reply]
Thanks for the replies!
